packages/features/handleMarkNoShow.ts

```diff
@@ -0,0 +1,126 @@
+import { prisma } from "@calcom/prisma";
+import { WebhookTriggerEvents } from "@calcom/prisma/client";
+import type { TNoShowInputSchema } from "@calcom/trpc/server/routers/publicViewer/noShow.schema";
+
+const getResultPayload = async (attendees: { email: string; noShow: boolean }[]) => {
+  if (attendees.length === 1) {
+    const [attendee] = attendees;
+    return {
+      message: attendee.noShow ? "x_marked_as_no_show" : "x_unmarked_as_no_show",
+      attendees: [attendee],
+    };
+  }
+  return { message: "no_show_updated", attendees: attendees };
+};
+
+const logFailedResults = (results: PromiseSettledResult<any>[]) => {
+  const failed = results.filter((x) => x.status === "rejected") as PromiseRejectedResult[];
+  if (failed.length < 1) return;
+  const failedMessage = failed.map((r) => r.reason);
+  console.error("Failed to update no-show status", failedMessage.join(","));
+};
+
+const handleMarkNoShow = async ({ bookingUid, attendees }: TNoShowInputSchema) => {
+  try {
+    const attendeeEmails = attendees?.map((attendee) => attendee.email) || [];
+    if (attendees && attendeeEmails.length > 0) {
+      const allAttendees = await prisma.attendee.findMany({
+        where: {
+          AND: [
+            {
+              booking: {
+                uid: bookingUid,
+              },
+              email: {
+                in: attendeeEmails,
+              },
+            },
+          ],
+        },
+        select: {
+          id: true,
+          email: true,
+        },
+      });
+
+      const allAttendeesMap = allAttendees.reduce((acc, attendee) => {
+        acc[attendee.email] = attendee;
+        return acc;
+      }, {} as Record<string, { id: number; email: string }>);
+
+      const updatePromises = attendees.map((attendee) => {
+        const attendeeToUpdate = allAttendeesMap[attendee.email];
+        if (!attendeeToUpdate) return;
+        return prisma.attendee.update({
+          where: { id: attendeeToUpdate.id },
+          data: { noShow: attendee.noShow },
+        });
+      });
+
+      const results = await Promise.allSettled(updatePromises);
+      logFailedResults(results);
+
+      const _attendees = results
+        .filter((x) => x.status === "fulfilled")
+        .map((x) => (x as PromiseFulfilledResult<{ noShow: boolean; email: string }>).value)
+        .map((x) => ({ email: x.email, noShow: x.noShow }));
+
+      const payload = await getResultPayload(_attendees);
+      const booking = await prisma.booking.findUnique({
+        where: { uid: bookingUid },
+        select: {
+          id: true,
+          eventType: {
+            select: {
+              id: true,
+              teamId: true,
+              userId: true,
+            },
+          },
+        },
+      });
+      const orgId = await getOrgIdFromMemberOrTeamId({
+        memberId: booking?.eventType?.userId,
+        teamId: booking?.eventType?.teamId,
+      });
+      const webhooks = await new WebhookService({
```

Comment:
@hariombalhara zomars added this code because constructor returns this async function.

```
  constructor(options: Parameters<typeof getWebhooks>[0]) {
    return (async (): Promise<WebhookService> => {
      this.options = options;
      this.webhooks = await getWebhooks(options);
      return this;
    })() as unknown as WebhookService;
  }
  ```

Reply:
Wow, that's the first time I am seeing async constructor. I am not sure if it is working as per expectation but I have never seen constructor doing async work.

Also, eslint is warning about it. But as it is existing code I am fine with it for now.